exporter/elasticsearchexporter/data_stream_router.go

```diff
@@ -38,6 +41,14 @@ func routeWithDefaults(defaultDSType, defaultDSDataset, defaultDSNamespace strin
 				return fmt.Sprintf("%s%s%s", prefix, fIndex, suffix)
 			}
 		}
+
+		const otelSuffix = ".otel"
+		// The naming convension for datastream is expected to be the following
+		// "logs-[dataset].otel-[namespace]"
+		if otel && !strings.HasSuffix(dataset, otelSuffix) {
```

Comment:
This is going to append `.otel` to `data_stream.dataset` set explicitly via attributes,  e.g. `data_stream.dataset=foo` will be sent to `data_stream.dataset=foo.otel`, while I was thinking about only changing only the defaultDSDataset in otel mode. @felixbarny what would be the better behavior in your opinion? If we are going with appending `.otel` like "logs-[dataset].otel-[namespace]", we should document it in user-facing docs.

Reply:
Per the [spec](https://github.com/elastic/opentelemetry-dev/blob/main/docs/design-decisions/ingest/mapping.md#data-stream-attributes), we'll always append `.otel`. So if there's an attribute called `foo.otel`, it will be changed to `foo.otel.otel`.

Reply:
Sounds good to me. @aleksmaus can you adjust the implementation and document that in README?

Reply:
Will do.